gdx/src/com/badlogic/gdx/utils/ArrayMap.java

```diff
@@ -398,6 +398,42 @@ public void truncate (int newSize) {
 		size = newSize;
 	}
 
+	public int hashCode () {
+		K[] keys = this.keys;
+		V[] values = this.values;
+		int h = 0;
+		for (int i = 0, n = size; i < n; i++) {
+			K key = keys[i];
+			V value = values[i];
+			if (key != null) h += key.hashCode() * 31;
+			if (value != null) h += value.hashCode();

```

Comment:
It seems the best way to handle this (including all other of our maps) is:

```java
h += (key != null ? key.hashCode() : 0) ^ (value != null ? value.hashCode() : 0);
```

See [MapEntry](https://svn.apache.org/repos/asf/harmony/enhanced/java/trunk/classlib/modules/luni/src/main/java/java/util/MapEntry.java) (which is called by [AbstractMap](https://svn.apache.org/repos/asf/harmony/enhanced/java/trunk/classlib/modules/luni/src/main/java/java/util/AbstractMap.java)).